src/v/pandaproxy/rest/handlers.cc

```diff
@@ -66,6 +66,19 @@ using server = proxy::server;
 
 } // namespace
 
+struct pp_parsing_error {
+    pp_parsing_error() = default;
+    explicit pp_parsing_error(std::string_view msg)
+      : msg(msg) {}
+    std::string_view msg;
+    [[noreturn]] [[gnu::cold]] void conversion_error() {
+        if (msg.empty()) {
+            msg = "parameter contained invalid control character";
```

Comment:
Should this be
```suggestion
            msg = "parameter is empty";
```

Or something like that?

Reply:
Sorry this probably isn't clear (and I'll add a comment to this effect).  The purpose of `msg` is so the caller can specify what the message will contain in the response to the user.  If one isn't provided this is the default value.  So really I should just set `msg` to some default value and remove this confusing conditional.